test/threads.jl

```diff
@@ -141,3 +141,40 @@ end
 @test_throws TypeError Atomic{Bool}
 @test_throws TypeError Atomic{BigInt}
 @test_throws TypeError Atomic{Float64}
+
+# Test atomic fences
+type CommBuf
+    data1::Atomic{Int}
+    data2::Atomic{Int}
+    correct::Bool
+    CommBuf() = new()
+end
+function test_fence_write(commbuf::CommBuf)
+    for i in 10000:-1:1
+        commbuf.data1[] = i
+        fence()
+        commbuf.data2[] = i
```

Comment:
Does this fail without the fence? IIUC, since the store to `data2` has a release order, the store to `data1` in the same loop has to be visible when the store to `data2` is visible.